sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubAsyncServiceClient.java

```diff
@@ -0,0 +1,895 @@
+import com.azure.messaging.webpubsub.models.GetAuthenticationTokenOptions;
+import com.azure.messaging.webpubsub.models.WebPubSubAuthenticationToken;
+import com.azure.messaging.webpubsub.models.WebPubSubContentType;
+import com.azure.messaging.webpubsub.models.WebPubSubPermission;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import static com.azure.core.util.tracing.Tracer.AZ_TRACING_NAMESPACE_KEY;
+import static com.azure.core.util.FluxUtil.withContext;
+import static com.azure.messaging.webpubsub.models.WebPubSubContentType.APPLICATION_JSON;
+import static com.azure.messaging.webpubsub.models.WebPubSubContentType.APPLICATION_OCTET_STREAM;
+import static com.azure.messaging.webpubsub.models.WebPubSubContentType.TEXT_PLAIN;
+
+/**
+ * The asynchronous client for connecting to an Azure Web Pub Sub hub (for a synchronous API, refer to the
+ * {@link WebPubSubServiceClient} class documentation). To create an instance of this class, refer to the code snippet
+ * below, and for more information about configuration options, refer to the JavaDoc for {@link WebPubSubClientBuilder}.
+ *
+ * <p>The Azure Web Pub Sub client instance must be connected to a
+ * {@link WebPubSubClientBuilder#hub(String) specific hub} that is represented by a non-null and non-empty String.
+ * Within the Azure Web Pub Sub client, users may perform operations including:
+ *
+ * <ul>
+ *     <li>Sending messages to {@link #sendToAll(String) everyone in the hub},</li>
+ *     <li>Sending messages to a {@link #sendToUser(String, String) specific user} or
+ *     {@link #sendToConnection(String, String) connection},</li>
+ *     <li>{@link #removeUserFromAllGroups(String) Removing a user} from all groups,</li>
+ *     <li>{@link #closeConnection(String) Closing a connection} of a specific user</li>
+ *     <li>To check the existence of a {@link #checkUserExists(String) user}, a
+ *     {@link #checkConnectionExists(String) connection}, or a {@link #checkGroupExists(String) group},</li>
+ * </ul>
+ *
+ * <p>It is possible to connect to a specific group within a hub by calling
+ * {@link #getAsyncGroup(String)}, which will return a {@link WebPubSubAsyncGroup}. All operations performed
+ * on this Azure Web Pub Sub group client will take into account the hub and group in which it exists.</p>
+ *
+ * <p><strong>Code Samples</strong></p>
+ *
+ * {@codesnippet com.azure.messaging.webpubsub.webpubsubclientbuilder.connectionstring.async}
+ *
+ * @see WebPubSubClientBuilder
+ * @see WebPubSubServiceClient
+ * @see WebPubSubAsyncGroup
+ */
+@ServiceClient(
+    builder = WebPubSubClientBuilder.class,
+    isAsync = true,
+    serviceInterfaces = WebPubSubsImpl.WebPubSubsService.class
+)
+public final class WebPubSubAsyncServiceClient {
+    // TODO (jogiles) find the appropriate name
+    static final String TRACING_NAMESPACE_VALUE = "Microsoft.WebSubPubService";
```

Comment:
According to the [resource providers](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers) documentation, this should be `Microsoft.SignalRService`